crates/modelardb_server/src/storage/uncompressed_data_manager.rs

```diff
@@ -212,24 +212,21 @@ impl UncompressedDataManager {
 
         // Prepare the timestamp column for iteration.
         let timestamp_index = model_table_metadata.timestamp_column_index;
-        let timestamp_column_array: &TimestampArray = data_points
-            .column(timestamp_index)
-            .as_any()
-            .downcast_ref()
-            .unwrap();
+        let timestamp_column_array: &TimestampArray =
```

Comment:
Maybe remove the type hint since it seems that we do not use it elsewhere when we use `array!`.

Reply:
I have removed most of the annotations, but two `Vec<_>` are needed so the compiler which data structures to collect arrays into.